client/src/store/utils/reducerFunctions.js

```diff
@@ -14,7 +14,7 @@ export const addMessageToStore = (state, payload) => {
   return state.map((convo) => {
     if (convo.id === message.conversationId) {
       const convoCopy = { ...convo };
-      convoCopy.messages.push(message);
+      convoCopy.messages = [message, ...convoCopy.messages];
```

Comment:
Is this the fix for the broke conversation order? I think it worked, but I'm curious if the runtime of this will be bad since we're doing a clone and it probably have to  iterate over all the messages to clone. An alternative way you can approach this is just to keep your original solution and reverse the conversation before we returned to the user.

Reply:
My solution worked fine. I changed flex to "column-reverse" to reverse the display other of the messages. Moreover, in #2 where you want to find the id of the latest messages that have been read. so you do not have to do a reverse map. just use map and the first item is the latest message you map through. 

Please let me know what you think.

Reply:
The messages is sent from the server is in "DESC" order. so the latest message is in the beginning of the messages array. If you use push then new message message will be added in the end, which messing the order of the messages array. so the new message should be added at the start. so it would be fine there. And the only change is from "column-reverse" in Messages.js file